transport/http/src/test/java/org/kaazing/gateway/transport/http/connector/specification/rfc7230/MessageFormatIT.java

```diff
@@ -17,87 +17,616 @@
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
+                will(new CustomAction("Latch countdown") {
+                    @Override
+                    public Object invoke(Invocation invocation) throws Throwable {
+                        closed.countDown();
+                        return null;
+                    }
+                });
+            }
+        });
+
+        connector.connect("http://localhost:8080/", handler, new ConnectSessionInitializerOBSInHeader());
+        assertTrue(closed.await(2, SECONDS));
+
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({"inbound.on.receiving.field.with.length.larger.than.wanting.to.process.must.reply.with.4xx/response"})
+    public void inboundOnReceivingFieldWithLengthLargerThanWantingToProcessMustReplyWith4xx() throws Exception {
+        final IoHandler handler = context.mock(IoHandler.class);
+        final CountDownLatch closed = new CountDownLatch(1);
+
+        context.checking(new Expectations() {
+            {
+                oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                will(new CustomAction("Latch countdown") {
+                    @Override
+                    public Object invoke(Invocation invocation) throws Throwable {
+                        closed.countDown();
+                        return null;
+                    }
+                });
+            }
+        });
+
+        connector.connect("http://localhost:8080/", handler, new ConnectSessionInitializerGetLongHeader());
+        assertTrue(closed.await(2, SECONDS));
+
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({"server.should.send.501.to.unknown.transfer.encoding/response"})
+    public void serverShouldSend501ToUnknownTransferEncoding() throws Exception {
+        final IoHandler handler = context.mock(IoHandler.class);
+        final CountDownLatch closed = new CountDownLatch(1);
+
+        context.checking(new Expectations() {
+            {
+                oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
+                will(new CustomAction("Latch countdown") {
+                    @Override
+                    public Object invoke(Invocation invocation) throws Throwable {
+                        closed.countDown();
+                        return null;
+                    }
+                });
+            }
+        });
+
+        connector.connect("http://localhost:8080/", handler, new ConnectSessionInitializerTransferEncoding());
+        assertTrue(closed.await(2, SECONDS));
+
+        k3po.finish();
+    }
+
+    @Ignore("Assertion Error: Unexpected Invocation")
```

Comment:
This should be treated the same as my earlier comment: if it can not be fixed, file a bug and add the link here, or remove it if it nothing can be done.